resources/experimental-feature.md

```diff
@@ -0,0 +1,89 @@
+
+## Some warnings about the usage
+
+- Experimental features are not recommended to be used in production.
+- For some provided experimental features, re-indexing will be needed. For this, more than only restarting the search engine, using a dump will be needed to enforce re-indexing.
+This means, in some situations, using a dump might be needed to activate an experimental feature but also to deactivate it. The users will be clearly informed in the related discussion how to activate the feature
+
+> ⚠️ When deactivating this kind of experimental feature, since our process is not mature yet, users can have some issues with their DB, even when using a dump.<br>
+> We recommend users always save their data (with snapshots and/or dumps) before starting to activate experimental features. Of course, the safest method is to use a script to feed Meilisearch.
+
+## Technical explanations
+
+### Why does Meilisearch need to be restarted (by using the `experimental` CLI flag)?
+
+Meilisearch uses LMDB to store the data, not only user documents but also a lot of internal data, like Meilisearch tasks. With LMDB, if we want to add or remove data structures to store this data, we need to close the LMDB environment and re-open it, so to restart Meilisearch.
+
+It means if an experimental feature implementation involves a modification of the data structures, users must restart Meilisearch; we cannot make it enable/disable from the HTTP routes.
+
+Unfortunately, this might concern most of the future features. That’s why we chose, for the moment, only to suggest flag CLI in order to enable/disable experimental features. However, this can be changed in the future or adapted to the situation.
+
+### Why could we need to use a dump?
+
+Since we need to restart Meilisearch, and changes happened on the DB side, an issue could happen when Meilisearch tries to read the “old” data.ms.
+
+Indeed, Meilisearch could try to read the old DB, but does not succeed to do it (crash), or do it badly (inconsistent results). That’s why we ask for dumps between 2 minor versions (there are often a lot of changes in the DB), and that’s why users sometimes will need to use a dump when passing from stable to experimental. The contrary (from experimental to stable) is riskier, even when using a dump, since we are not mature enough yet, but we will be safer and safer in the future.
+
+### Restarting Meilisearch and using dump is not convenient; could we improve this in the future?
+
+Indeed, we know it’s not convenient for the users. We will always try to ask to use a dump only when it’s necessary.
+
+Regarding the restart, we will do our best to try removing it, and then avoid using the dump at the same time. However, it will require a lot of implementation time we don’t have the time currently to spend. But depending on the situation, we might want to put effort into it. We know it will facilitate the exposition of the experimental feature on the Cloud side for example.
```

Comment:
The bit regarding Meilisearch Cloud seemed unnecessary in the context of this document. Let me know if you think it's really important.

```suggestion
We understand the situation is inconvenient and less than ideal. We will only ask users to use dumps when activating experimental features when it’s strictly necessary.

Avoiding restarts is more difficult, especially for features that currently require database migrations with dumps. We are not currently working on this, but the situation might change in the future.
```